ci3/cache/download

```diff
@@ -1,41 +1,29 @@
 #!/bin/bash
-set -eu
+set -eu -o pipefail
+[ "${BUILD_SYSTEM_DEBUG:-}" = 1 ] && set -x
 
 if [ "$#" -lt 1 ]; then
   echo "Usage: $0 <tar.gz_file_to_download_and_extract>" >&2
   exit 1
 fi
 
-if [ "${USE_CACHE:-0}" -lt 1 ] ; then
+if [ "${USE_CACHE:-0}" -lt 1 ]; then
   # Only download if USE_CACHE is 1
+  echo "Not using cache for $1 because USE_CACHE=0."
```

Comment:
Think we need `>&2` to print to stderr.
Reason being that noir-contracts actually uses stdout to pass results back to parallel. So this could get mixed in i think.
It's a bit of a footgun, the other option is that noir-contracts redirects stdout to dev/null for the cache calls, but then you can't see the output for debugging.
I think we should just adopt the pattern that all ci3 related output should be going to stderr.

Reply:
this is probably far too brittle to rely on though - so it makes more sense that calling code that wants to use stdout, redirect stdout of ci scripts to stderr themself, because at some point someones going to do the "non idiomatic thing" of writing log lines to stdout - or use some program that doesn't-do-the-right-thing.

Still, we can try and be idiomatic unix. I went looking for reassurance but GPT seems to be being quite curt with me today...
![image](https://github.com/user-attachments/assets/e4b5ab2d-a67f-4021-8463-677d5e397d36)